### PR TITLE
Updated Filter logic

### PR DESCRIPTION
**Pull request by flacoman91**
Updated logic for filters with Parent/Child relationships (issue/product)

Check only parent when all available children are checked
Uncheck parent and apply the rest of the siblings when a parent is checked, and a child item is "unchecked"

Fake "Checked" subitems appear on the front-end, but do not apply it in the query or in redux

Don't apply fake pills

### CODE REVIEW

File: src/components/Filters/AggregationItem.jsx [hunk 1]
```diff
@@ -40,24 +51,82 @@ AggregationItem.propTypes = {
...
 
 export const mapStateToProps = ( state, ownProps ) => {
-  const field = state.query[ownProps.fieldName]
+  const aggs = coalesce( state.aggs, ownProps.fieldName, [] )
+  const filters = coalesce( state.query, ownProps.fieldName, [] )
   const value = ownProps.item.key
-  const active = typeof field !== 'undefined' && field.indexOf( value ) > -1
-  return { active }
+  const parentKey = value.split( SLUG_SEPARATOR )[0]
+  const active = filters.includes( value ) || filters.includes( parentKey )
```
Comment (JeffreyMFarley) [inline src/components/Filters/AggregationItem.jsx:66]: clever

File: src/components/Filters/__tests__/AggregationBranch.spec.jsx [hunk 2]
```diff
@@ -127,21 +127,32 @@ describe('component::AggregationBranch', () => {
       const checkbox = target.find('li.parent input[type="checkbox"]')
       checkbox.simulate('change')
       expect(props.uncheckParent).not.toHaveBeenCalled()
-      expect(props.checkParent).toHaveBeenCalledWith(
-        'issue', ['foo', 'foo•bar', 'foo•baz', 'foo•qaz', 'foo•quux']
-      )
+      expect(props.checkParent).toHaveBeenCalled()
```
Comment (JeffreyMFarley) [inline src/components/Filters/__tests__/AggregationBranch.spec.jsx:130]: Are you not able to test with expected values anymore?

Comment (flacoman91) [reply src/components/Filters/__tests__/AggregationBranch.spec.jsx:130]: the function is passing in props, so it's also adding in some js functions so it looks a bit cluttered in this test.  i can explicitly check for these

File: src/utils/__tests__/compare.spec.jsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+// ----------------------------------------------------------------------------
+// Tests
+describe( 'arrayEquals', () => {
+  it( 'compares different length arrays', () => {
+    const res = sut.arrayEquals( [], [ 1, 2 ] )
+    expect( res ).toBeFalsy()
+  } )
+
+  it( 'compares equal arrays', () => {
+    const res = sut.arrayEquals( [ 1, 2 ], [ 1, 2 ] )
```
Comment (JeffreyMFarley) [inline src/utils/__tests__/compare.spec.jsx:12]: Does element order matter?

Comment (flacoman91) [reply src/utils/__tests__/compare.spec.jsx:12]: yes. I just added another test to verify a different order.

**Comment by sephcoster** [comment]
Great work, thanks!

**Pull request merged**